vtex/faststore
# add Delivery Mode

## What's the purpose of this pull request?

Create a smooth navigation with address and logistics set through FastStore.

## How it works?

We set the address and logisticsInfo on the OrderForm according to the user's session.

## How to test it?

This PR has several //remove before merge comments to facilitate testing.

Just use CEP: 60810-350 to regionalize through the Region Modal that we already have a pick up point created in the store framework.

After regionalizing, just continue with the flow to the checkout and see that the pre-selection made at the pick up point is maintained. This flow is important for Grocery customers mainly.

## References

[[[[](https://help.vtex.com/en/tutorial/pickup-points--2fljn6wLjn8M4lJHA6HP3R#setup)](https://storeframework.myvtex.com/admin/pickup-points/100-200-300-400/)](https://storeframework.myvtex.com/admin/app/shipping-strategy/shipping-policy?id=1dbdfb3)](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/attachments/shippingData)

cURL of example:

```
curl --location 'https://storeframework.myvtex.com/api/checkout/pub/orderForm/{{orderFormId}}/attachments/shippingData' \
--data '{
    "logisticsInfo": [
        {
            "addressId": null,
            "itemIndex": 0,
            "selectedDeliveryChannel": "pickup-in-point",
            "selectedSla": "pickuppoint (100-200-300-400)"
        }
    ],
    "clearAddressIfPostalCodeNotFound": false,
    "selectedAddresses": [
        {
            "addressId": null,
            "addressType": "search",
            "country": "BRA",
            "postalCode": "60810-350"
        }
    ]
}'
```
